controllers/clusterversion_controller.go

```diff
@@ -234,6 +287,28 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	prometheusRule, err := parsePrometheusRule(pvcPrometheusRules)
+	if err != nil {
+		c.log.Error(err, "Unable to retrieve prometheus rules.", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+		return ctrl.Result{}, err
+	}
+	prometheusRule.SetNamespace(c.OperatorNamespace)
+	if err := c.own(prometheusRule); err != nil {
+		return ctrl.Result{}, err
+	}
+	operatorConfig, err := c.getOperatorConfig()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	applyLabels(operatorConfig.Data["OCS_METRICS_LABELS"], &prometheusRule.ObjectMeta)
+
+	if err := c.createOrUpdatePrometheusRule(prometheusRule); err != nil {
```

Comment:
We should be able to use controller-runtime client here and avoid all the setup code below.

Reply:
i used the code from ocs-operator; any specific reason not to use the controller-runtime client in OCS-operator?

Reply:
Not really. It's just old code that should be updated.